docs/design/security_context.md

```diff
@@ -0,0 +1,158 @@
+# Security Contexts
+## Abstract
+A security context is a set of constraints that are applied to a container in order to achieve the following goals (from [security design](security.md)):
+
+1.  Ensure a clear isolation between container and the underlying host it runs on
+2.  Limit the ability of the container to negatively impact the infrastructure or other containers
+
+## Background
+
+The problem of securing containers in Kubernetes has come up [before](https://github.com/GoogleCloudPlatform/kubernetes/issues/398) and the potential problems with container security are [well known](http://opensource.com/business/14/7/docker-security-selinux). Although it is not possible to completely isolate Docker containers from their hosts, new features like [user namespaces](https://github.com/docker/libcontainer/pull/304) make it possible to greatly reduce the attack surface.
+
+## Motivation
+
+### Container isolation
+
+In order to improve container isolation from host and other containers running on the host, containers should only be 
+granted the access they need to perform their work. To this end it should be possible to take advantage of Docker 
+features such as the ability to [add or remove capabilities](https://docs.docker.com/reference/run/#runtime-privilege-linux-capabilities-and-lxc-configuration) and [assign MCS labels](https://docs.docker.com/reference/run/#security-configuration) 
```

Comment:
Everything - the container would be relabeled, the process would have those labels, and any volumes would either be labelled or potentially left as is (in a few cases maybe this is reasonable?).  Common case though is "you get these labels".  I believe we have all but the volume support upstream and we carry the relabeling support on RHEL docker.


> On Jan 29, 2015, at 7:49 PM, Eric Tune <notifications@github.com> wrote:
> 
> In docs/design/security_context.md:
> 
> > +A security context is a set of constraints that are applied to a container in order to achieve the following goals (from [security design](security.md)):
> > +
> > +1.  Ensure a clear isolation between container and the underlying host it runs on
> > +2.  Limit the ability of the container to negatively impact the infrastructure or other containers
> > +
> > +## Background
> > +
> > +The problem of securing containers in Kubernetes has come up [before](https://github.com/GoogleCloudPlatform/kubernetes/issues/398) and the potential problems with container security are [well known](http://opensource.com/business/14/7/docker-security-selinux). Although it is not possible to completely isolate Docker containers from their hosts, new features like [user namespaces](https://github.com/docker/libcontainer/pull/304) make it possible to greatly reduce the attack surface.
> > +
> > +## Motivation
> > +
> > +### Container isolation
> > +
> > +In order to improve container isolation from host and other containers running on the host, containers should only be 
> > +granted the access they need to perform their work. To this end it should be possible to take advantage of Docker 
> > +features such as the ability to [add or remove capabilities](https://docs.docker.com/reference/run/#runtime-privilege-linux-capabilities-and-lxc-configuration) and [assign MCS labels](https://docs.docker.com/reference/run/#security-configuration) 
> What types of volumes would the MCS labels be used with? Presumably there aren't files that are sensitive for the container process in the emptydir. If this for files in the hostDir, or some other type of volume?
> 
> —
> Reply to this email directly or view it on GitHub.
>